src/v/cluster/partition.h

```diff
@@ -124,6 +129,7 @@ class partition {
 private:
     consensus_ptr _raft;
     std::unique_ptr<raft::log_eviction_stm> _nop_stm;
+    std::unique_ptr<raft::id_allocator_stm> _id_allocator_stm;
```

Comment:
I think this is not a good place for the `_id_allocator_stm`. Partition is a generic structure, id allocator is something very specific, maybe it would be better to make the `_id_allocator_stm`  owned by i`d_allocator_service`

Reply:
That's true, but there is also another reasoning which justifies current design. State machines are coupled with the logs (consesnus.h) they run on for example we should stop a stm before we stop a consensus. In that sense if we put them apart it creates a possibility for an error and forces us to introduce a lot of infrastructure to keep them in sync. Also it forces us to write additional code to rebalance the id_allocator_stm's service when we rebalance partitions - right now we get it for free. It seems that keeping the stm there makes things simpler.

Reply:
@mmaslankaprv are fine with leaving id_allocator_stm in the partition? it helps us with managing lifetime and prevents from writing extra code to deal with partition rebalancing